arch/xtensa/src/esp32s3/esp32s3_gpio.c

```diff
@@ -171,24 +237,31 @@ void esp32s3_gpio_matrix_in(uint32_t gpio, uint32_t signal_idx, bool inv)
  * Name: esp32s3_gpio_matrix_out
  *
  * Description:
- *   Set signal output to gpio
- *   NOTE: one signal can output to several gpios
- *   If signal_idx == 0x100, cancel output put to the gpio
+ *   Set signal output to GPIO.
+ *   NOTE: one signal can output to several GPIOs.
+ *
+ * Input Parameters:
+ *   pin           - GPIO pin to be configured.
+ *   signal_idx    - Signal index.
+ *                   - If signal_idx == 0x100, cancel output to the GPIO.
+ *   out_inv       - Flag indicating whether the signal output is inverted.
+ *   oen_inv       - Flag indicating whether the signal output enable is
+ *                   inverted.
+ *
+ * Returned Value:
+ *   None.
  *
  ****************************************************************************/
 
-void esp32s3_gpio_matrix_out(uint32_t gpio, uint32_t signal_idx,
-                             bool out_inv, bool oen_inv)
+void esp32s3_gpio_matrix_out(uint32_t pin, uint32_t signal_idx, bool out_inv,
+                             bool oen_inv)
 {
-  uint32_t regaddr = GPIO_FUNC0_OUT_SEL_CFG_REG + (gpio * 4);
+  uint32_t regaddr = GPIO_FUNC0_OUT_SEL_CFG_REG + (pin * 4);
   uint32_t regval = signal_idx << GPIO_FUNC0_OUT_SEL_S;
 
-  if (gpio >= ESP32S3_NGPIOS)
-    {
-      return;
-    }
+  DEBUGASSERT(is_valid_gpio(pin));
 
-  putreg32(1ul << gpio, GPIO_ENABLE_W1TS_REG);
+  putreg32(1ul << pin, GPIO_ENABLE_W1TS_REG);
```

Comment:
```suggestion
  putreg32(UINT32_C(1) << pin, GPIO_ENABLE_W1TS_REG);
```

Reply:
do we need to have something like 32 check? or `% 32`?

Reply:
Actually, there is actually another register for pins whose number are greater than 31.
I need to double check it, but I think the current driver seems to not handle this case.
Thanks for the heads up.

Reply:
@pkarashchenko Indeed, the handling for that scenario was missing. I've also applied the fix you suggested. Thanks!